include/seqan3/alignment/pairwise/edit_distance_algorithm.hpp

```diff
@@ -0,0 +1,82 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::edit_distance_algorithm.
+ * \author Marcel Ehrhardt <marcel.ehrhardt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/pairwise/edit_distance_unbanded.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief This algorithm unifies different edit distance implementations and uses the appropriate one depending on the
+ * given configuration.
+ * \implements std::Invocable
+ * \tparam config_t The configuration type.
+ * \tparam traits_t The traits type.
+ *
+ * \details
+ *
+ * This wrapper class is used to decouple the sequence types from the algorithm class type.
+ * Within the alignment configuration a std::function object with this wrapper stored is returned
+ * if an edit distance should be computed. On invocation it delegates the call to the actual implementation
+ * of the edit distance algorithm, while the interface is unified with the execution model of the pairwise alignment
+ * algorithms.
+ */
+template <typename config_t, typename traits_t>
+class edit_distance_algorithm
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr edit_distance_algorithm() = default; //!< Defaulted
```

Comment:
align me

Reply:
Why? :sad: This will need extra work, if someone changes the name :sob: